Sources/SwiftProtobufCore/BinaryEncoder.swift

```diff
@@ -28,8 +27,8 @@ internal struct BinaryEncoder {
         pointer = pointer.advanced(by: 1)
     }
 
-    private mutating func append(contentsOf data: Data) {
-        data.withUnsafeBytes { dataPointer in
+    private mutating func append<Bytes: SwiftProtobufContiguousBytes>(contentsOf bytes: Bytes) {
```

Comment:
Does this still need to be generic?

Reply:
Yes, it's used by `BinaryEncoder/putBytesValue(value:)`, which is used by `BinaryDelimited/serialize(message:to stream:partial:)` and uses `[UInt8]`. However other uses of `append(contentsOf:)` use `Data`, so we're using both

Reply:
Actually, scratch all that - I'll revert this to being non-generic. I can just use Data in BinaryDelimited and the TextFormatEncoding visitor. We'll be reading Data fields from the protobufs themselves anyways, and will have to deserialise them into Data when converting back to protobuf, so it doesn't make sense for this to be generic.

Reply:
Looking at this again - I've reverted some of the generics in other methods, but did put it back here. I decided to still use `SwiftProtobufContiguousBytes` instead of `Data` in `BinaryDelimited/serialize` to give the user more flexibility, since the method serialises multiple `Message`s into an `OutputStream`. I had missed that here we're not encoding a field, but rather reusing the `putBytesValue` method of the `BinaryEncoder` to encode the **whole** `Message`.